flink-streaming-java/src/main/java/org/apache/flink/streaming/api/graph/StreamGraphGenerator.java

```diff
@@ -266,6 +266,12 @@ private void configureStreamGraph(final StreamGraph graph) {
 		graph.setJobName(jobName);
 
 		if (shouldExecuteInBatchMode) {
+
+			if (checkpointConfig.isCheckpointingEnabled()) {
+				LOG.info("Disabled Checkpointing. Checkpointing is not supported when executing jobs in BATCH mode.");
+				checkpointConfig.disableCheckpointing();
```

Comment:
I don't like that we're setting it on the `CheckpointConfig` that the user gave to us. I think it's in general not good that we just pass this through but now is the first time that we modify it here, I think.

Maybe we should do an internal copy of the checkpoint config and then we can modify it?

Reply:
I agree. The only thing I am afraid is if at any point we are taking again the checkpoint config from the `env` and we disregard what the `StreamGraphGenerator` has set. From a quick look, it does not seem to be the case, so I pushed another commit with the change. Please have a look and let me know what you think.